pkg/image/api/types.go

```diff
@@ -60,6 +77,30 @@ type ImageLayer struct {
 	Size int64
 }
 
+// SignatureValidationResult holds a result of signature validation.
+type SignatureValidationResult string
+
+// ImageSignature holds a signature of an image.
+type ImageSignature struct {
+	// Describes a type of stored blob.
+	Type string
+	// An opaque binary string which is an image's signature.
+	Content []byte
+	// A time of last server-side validation. IOW a time of the last update to ValidationResult.
+	LastValidation *unversioned.Time
```

Comment:
Sounds good to me.
On 05/27/2016 02:29 PM, David Eads wrote:
>
> In pkg/image/api/types.go <https://github.com/openshift/origin/pull/8371#discussion_r64897252>:
>
> > @@ -60,6 +77,30 @@ type ImageLayer struct {
> >  	Size int64
> >  }
> >
> > +// SignatureValidationResult holds a result of signature validation.
> > +type SignatureValidationResult string
> > +
> > +// ImageSignature holds a signature of an image.
> > +type ImageSignature struct {
> > +	// Describes a type of stored blob.
> > +	Type string
> > +	// An opaque binary string which is an image's signature.
> > +	Content []byte
> > +	// A time of last server-side validation. IOW a time of the last update to ValidationResult.
> > +	LastValidation *unversioned.Time
>
>     I wanted to emphasize that this field is tightly related to ValidationResult. If renamed to Checked, then the
>     second field should be renamed to CheckResult which seems ambiguous to me.
>
> Thing is, the signature hasn't been validated with respect to the image its attached to. It's been parsed and the 
> issuer was probably checked, but the using validation here could pretty easily be confused.
>
> |Checked| and |SigningCertState|?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/openshift/origin/pull/8371/files/e2ee2502036892b3ee92f2be36e2f3287bb7d628#r64897252>, or mute the 
> thread <https://github.com/notifications/unsubscribe/ABaQkwbvFVXLXlwotj6j2sx4IXv3eAXaks5qFuOkgaJpZM4IBAFQ>.
>